pkg/crc/preflight/preflight_check_tray_darwin.go

```diff
@@ -127,48 +105,111 @@ func fixTrayExistsAndRunning() error {
 		StdOutFilePath: stdOutFilePathDaemon,
 	}
 
-	trayConfig := AgentConfig{
-		BinaryPath:     filepath.Join(constants.CrcBinDir, constants.TrayBinaryName, "Contents", "MacOS", "CodeReady Containers"),
-		StdOutFilePath: stdOutFilePathTray,
-	}
 	logging.Debug("Creating daemon plist")
 	err = createPlist(daemonPlistTemplate, daemonConfig, daemonPlistFilePath)
 	if err != nil {
 		return err
 	}
+	// load plist and restart agent
+	if err := launchctlLoadPlist(daemonPlistFilePath); err != nil {
+		return err
+	}
+
+	if err := restartAgent(daemonAgentLabel); err != nil {
```

Comment:
I'm not sure `launchtlLoadPlist` and `restartAgent` should be done here. We created the plist because it did not exist, our job is done. Another preflight check will load/start the agent if it's needed. If the plist was missing initially, there is nothing to *re*start anyway, is there?

Reply:
Yes, we only need to restart in case of an upgrade, so this is only needed in `fixTrayVersion`